src/libmongoc/src/mongoc/mongoc-write-command.c

```diff
@@ -1355,6 +1357,22 @@ _mongoc_write_result_merge (mongoc_write_result_t *result,   /* IN */
       result->n_writeConcernErrors++;
    }
 
+   /* If a server error ocurred, then append the raw response to the
+    * error_replies array. */
```

Comment:
Offhand, would the first `{ ok: 0 }` command response (post-retry) completely interrupt a bulk write operation regardless of the `ordered` option (note: `ordered: false` is essentially continue-on-error).

Our specs are ambiguous about this (see: [DRIVERS-2090](https://jira.mongodb.org/browse/DRIVERS-2090), [DRIVERS-2159](https://jira.mongodb.org/browse/DRIVERS-2159)).

I noted in the unified test runner code below, the assertion is only done on `errorReplies.0`, which assumes that three will only ever be on error reply. That's fine, as I intentionally wrote the tests with the above ambiguity in mind (see: [mongodb/specifications@33a03f2](https://github.com/mongodb/specifications/blob/33a03f2eae1f479165fbde97993afb18241204af/source/crud/tests/unified/bulkWrite-errorResponse.yml#L19).

But here, I'm more curious about whether multiple `errorReplies` are even possible in libmongoc. If not, are we just using an `errorReplies` array on the result for flexibility?

Reply:
It is possible for libmongoc to return multiple `errorReplies`. Here is a [test](https://github.com/mongodb/mongo-c-driver/pull/1144/commits/f9ece40cb73db9921873e762e641be6d3e628e70) displaying this behavior. Since `ordered: false` continues on error when the `failPoint` is hit twice, two `errorReplies` are returned by libmongoc, as well as other errors that occurred if they did (in the test a `writeError` also was raised).

Reply:
Thanks for indulging my question. Looks like an oversight in the test runner syntax I introduced in [DRIVERS-2385](https://jira.mongodb.org/browse/DRIVERS-2385). I've made a note about this in [DRIVERS-1547](https://jira.mongodb.org/browse/DRIVERS-1547?focusedCommentId=4981812&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-4981812) as that's the most likely ticket we'd use to add spec test for multiple command failures in a bulk write.